tests/phpunit/tests/l10n/loadTextdomainJustInTime.php

```diff
@@ -276,6 +276,31 @@ public function test_get_locale_is_called_only_once_per_textdomain() {
 		$this->assertSame( 1, $this->locale_count );
 	}
 
+	/**
+	 * @ticket 37997
+	 * @ticket 39210
+	 *
+	 * @covers ::_load_textdomain_just_in_time
+	 */
+	public function test_get_locale_is_called_only_once_per_textdomain_with_custom_lang_dir() {
+		load_plugin_textdomain( 'custom-internationalized-plugin', false, dirname( plugin_basename( __FILE__ ) ) . '/languages' );
+
+		$textdomain = 'custom-internationalized-plugin';
+
+		add_filter( 'locale', array( $this, '_filter_locale_count' ) );
+
+		__( 'Foo', $textdomain );
+		__( 'Bar', $textdomain );
+		__( 'Baz', $textdomain );
+		__( 'Foo Bar', $textdomain );
+		__( 'Foo Bar Baz', $textdomain );
+
+		remove_filter( 'locale', array( $this, '_filter_locale_count' ) );
```

Comment:
See above. Additionally though, you can remove this function call in any case, since tests don't need to remove their added hooks, as that happens automatically.